analyses/snv-callers/scripts/04-create_consensus_mut_files.R

```diff
@@ -0,0 +1,278 @@
+vaf_df <- readr::read_rds(file.path(
+  opt$merged_files,
+  paste0("all_callers_vaf.", opt$file_format)
+))
+
+# If the caller chosen for the VAF reported isn't in the column, stop.
+if (!(opt$vaf %in% vaf_df$caller)) {
+  stop("Caller chosen with --vaf does not exist in the 'callers column in the
+  master VAF file.")
+}
+tmb_df <- readr::read_rds(file.path(
+  opt$merged_files,
+  paste0("all_callers_tmb.", opt$file_format)
+))
+callers_per_mutation <- readr::read_rds(file.path(
+  opt$merged_files,
+  paste0("callers_per_mutation.", opt$file_format)
+))
+
+# If the file exists or the overwrite option is not being used, do not write the
+# consensus mutation file.
+if (file.exists(consensus_mut_file) && !opt$overwrite) {
+  # Stop if this file exists and overwrite is set to FALSE
+  warning(cat(
+    "A consensus mutation file already exists: \n",
+    consensus_mut_file, "\n",
+    "Use --overwrite if you want to overwrite it."
+  ))
+} else {
+  # Let's get the mutation ids for combination of
+  consen_mutations <- callers_per_mutation %>%
+    dplyr::filter(caller_combo == dplyr::sym(opt$combo)) %>%
+    dplyr::select(-caller_combo) %>%
+    dplyr::pull(mutation_id)
+
+  # Stop if no mutations are found
+  if (length(consen_mutations) == 0) {
+    stop("No mutations had this combination of callers call it. Double check that
+       the combination you specified with --combo is spelled exactly right and
+       the callers are in alphabetical order. ")
+  }
+
+  # Isolate the mutations to only these mutations, use the Strelka2 stats.
+  consen_mutation <- vaf_df %>%
+    dplyr::filter(
+      caller == opt$vaf,
+      mutation_id %in% consen_mutations
+    ) %>%
+    readr::write_tsv(consensus_mut_file)
+}
+
+# If the file exists or the overwrite option is not being used, do not zip the
+# consensus mutation file.
+if (file.exists(consensus_mut_zip_file) && !opt$overwrite) {
+  # Stop if this file exists and overwrite is set to FALSE
+  warning(cat(
+    "A zipped consensus mutation file already exists: \n",
+    consensus_mut_zip_file, "\n",
+    "Use --overwrite if you want to overwrite it."
+  ))
+} else {
+  # Zip this file up.
+  zip(consensus_mut_zip_file, consensus_mut_file)
+}
+############################### Re-calculate TMB ###############################
+# If the file exists or the overwrite option is not being used, do not create
+# the consensus TMB file
+if (file.exists(consensus_tmb_file) && !opt$overwrite) {
+  # Stop if this file exists and overwrite is set to FALSE
+  warning(cat(
+    "A consensus TMB file already exists: \n",
+    consensus_tmb_file, "\n",
+    "Use --overwrite if you want to overwrite it."
+  ))
+} else {
+  # Set up BED region files for TMB calculations
+  wgs_bed <- readr::read_tsv(file.path(opt$bed_wgs), col_names = FALSE)
+  wxs_bed <- readr::read_tsv(file.path(opt$bed_wxs), col_names = FALSE)
+
+  # Calculate size of genome surveyed
```

Comment:
Can you put a detailed comment in here about the indices please? I see that there are not column names.